pkg/version/manager.go

```diff
@@ -23,19 +23,49 @@ import (
 	"github.com/Masterminds/semver/v3"
 
 	v1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
 	kubermaticlog "k8c.io/kubermatic/v2/pkg/log"
 	"k8c.io/kubermatic/v2/pkg/validation/nodeupdate"
 )
 
+// IncompatibilityCondition is the type defining the cluster or datacenter condition that must be met to block a specific version
+type IncompatibilityCondition string
+
+const (
+	// AlwaysCondition represent an always true matching condition used while checking provider incompatibilities
+	AlwaysCondition IncompatibilityCondition = "always"
+	// ExternalCloudProviderCondition is an incompatibility condition that represents the usage of the external Cloud Provider
+	ExternalCloudProviderCondition IncompatibilityCondition = "externalCloudProvider"
+)
+
+// OperationType is the type defining the operations triggering the compatibility check (CREATE or UPDATE)
+type OperationType string
+
+const (
+	// CreateOperation represents the creation of a new cluster
+	CreateOperation OperationType = "CREATE"
+	// UpdateOperation represents the update of an already existing cluster
+	UpdateOperation OperationType = "UPDATE"
```

Comment:
I find calling this `UPDATE` slightly confusing. As far as I get from the code, this is used to block cluster upgrades. Therefore, it might make sense to call it `UPGRADE` instead.